packages/cardpay-sdk/sdk/utils/safe-utils.ts

```diff
@@ -252,12 +298,75 @@ export async function executeSend(
   return response.json();
 }
 
+export async function generateCreate2SafeTx(
+  web3: Web3,
+  owners: string[],
+  treshold: number,
+  to: string,
+  data: string,
+  fallbackHandler: string,
+  paymentToken: string,
+  payment: string,
+  paymentReceiver: string,
+  saltNonce: number
+) {
+  let gnosisSafeProxyFactory = new web3.eth.Contract(
+    GnosisSafeProxyFactoryABI as AbiItem[],
+    await getAddress('gnosisProxyFactory_v1_3', web3)
+  );
+  let gnosisSafeMasterCopy = new web3.eth.Contract(
+    GnosisSafeABI as AbiItem[],
+    await getAddress('gnosisSafeMasterCopy', web3)
+  );
+
+  let initializer = gnosisSafeMasterCopy.methods
+    .setup(owners, treshold, to, data, fallbackHandler, paymentToken, payment, paymentReceiver)
+    .encodeABI();
+  let expectedSafeAddress = await calculateCreateProxyWithNonceAddress(
+    gnosisSafeProxyFactory,
+    gnosisSafeMasterCopy.options.address,
+    initializer,
+    saltNonce
+  );
+
+  let create2SafeData = gnosisSafeProxyFactory.methods
+    .createProxyWithNonce(gnosisSafeMasterCopy.options.address, initializer, saltNonce)
+    .encodeABI();
+  let create2SafeTx: Transaction = {
+    to: gnosisSafeProxyFactory.options.address,
+    value: '0',
+    data: create2SafeData,
+    operation: Operation.CALL,
+  };
+
+  return { expectedSafeAddress, create2SafeTx };
+}
+
+async function calculateCreateProxyWithNonceAddress(
+  gnosisSafeProxyFactory: Contract,
+  masterCopyAddress: string,
+  initializer: string,
+  saltNonce: number
```

Comment:
wont this be a problem as a number if we are using really large uint256 vlaues?

Reply:
right, will change it to string

Reply:
updated